src/main/java/sh/okx/railswitch/SetDestinationCommand.java

```diff
@@ -1,44 +1,80 @@
 package sh.okx.railswitch;
 
+import com.google.common.base.CharMatcher;
+import com.google.common.base.Strings;
+import java.util.ArrayList;
+import java.util.List;
 import org.bukkit.ChatColor;
 import org.bukkit.command.Command;
 import org.bukkit.command.CommandExecutor;
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
 
+/**
+ * The Set Destination command handler.
+ */
 public class SetDestinationCommand implements CommandExecutor {
-  private final RailSwitch plugin;
-
-  public SetDestinationCommand(RailSwitch plugin) {
-    this.plugin = plugin;
-  }
-
-  @Override
-  public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
-    if (!(sender instanceof Player)) {
-      return false;
-    }
-
-
-    Player player = (Player) sender;
-    if (args.length < 1) {
-      player.sendMessage(ChatColor.GREEN + "Unset rail destination (use /" + label + " <destination> to set your destination).");
-      plugin.getDatabase().removePlayerDestination(player);
-      return true;
-    }
-
-    String dest = String.join(" ", args);
-    if (!plugin.isValidDestination(dest)) {
-      player.sendMessage(ChatColor.RED + "Destinations can each not be more than 40 characters and may only use alphanumerical characters, ASCII symbols, and spaces.");
-      return true;
+    
+    private final RailSwitchPlugin plugin;
+    
+    /**
+     * Creates a new set destination command handler, passing through the current RailSwitch plugin instance.
+     *
+     * @param plugin The plugin to pass through to this command handler.
+     */
+    public SetDestinationCommand(RailSwitchPlugin plugin) {
+        this.plugin = plugin;
     }
-
-    long start = System.currentTimeMillis();
-    plugin.getDatabase().setPlayerDestination(player, dest);
-    if (plugin.isTimings()) {
-      plugin.getLogger().info("Set destination took " + (System.currentTimeMillis() - start) + "ms");
+    
+    @Override
+    public boolean onCommand(CommandSender sender, Command command, String label, String[] arguments) {
+        if (!(sender instanceof Player)) {
+            return false;
+        }
+        
+        Player player = (Player) sender;
+        
+        // Convert
+        List<String> destinations = new ArrayList<>();
+        checker:
+        for (String argument : arguments) {
+            // If the argument is null or empty, it's likely from a double space, just ignore it
+            if (Strings.isNullOrEmpty(argument)) {
+                continue;
+            }
+            if (argument.length() > 40 || !CharMatcher.inRange('0', '9').
```

Comment:
Should probably just use a single Regex here

Reply:
Done